languages/python/reference/exercise-concepts/markdown.md

```diff
@@ -0,0 +1,48 @@
+# Concepts for markdown
+
+[Example Implementation](https://github.com/exercism/python/blob/master/exercises/markdown/example.py)
+
+## General  
+
+
+- functions: This exercise requires a re-write of pre-esiting code that uses functions to parse passed-in text in _markdown_.  Tests expect a function named `parse` that can be called to transform the _markdown_ formatted text and return HTML formatted text.
+
+- function arguments:  The example solution uses multiple functions that take various function arguments to operate on different pieces of the passed in markdown string.
+
+- `re` module:  Both the original code to be refactored for this exercise and the example solution import and use the `re` module for Regular Expressions in python.
+
+- `import` key word/ importing:  Both the original code to be refactored for the exercise and the example solution use the `import` keyword to import the `re` module in support of Regular Expressions in python.
+
+- `str.format()` :  The example solution uses `str.split()` to break the passed in markdown string into a list of lines broken up by the `\n` character.
+
+- `re.match()`:  Method of the `re` class that returns a `match` object with any matched values from a specified Regualr Expression or pre-compliled Regular Expression.  The example uses `re.match()` in multiple places to search for text patterns that need re-formatting or subsitituting. 
+
+- regular expressions:  As a seperate language with a seperate interpreter that is often embeded inside another programming language.  Like many other programming lanagues in use, python supports the Regular Expression language and implements a Regular Expression interpreter in the core `re` module.  Both the original code to be refactored for this exercise and the example solution use the core `re` module to access Regualr Expression functionality.
+
+- `return` keyword:  Most of the functions in the example solution specify a _return_ value using the `return` keyword.
+
+- `None`:  None (null) type.  Used to define a null variable or an object.  It is in of itself a singleton in any given python program.  The example uses `None` as an expecit return value in several functions.
+
+- booleans:  True and False of type `bopl`.  The example solution uses `True` and `False` as return values from functions that test membership in a list of values.
+
+- assignment:  The example solution uses assingment for variables and other values.
+
+- `re.sub()`:  Metho of the `re` class that replaces a Regular Expression match with a new value.  The example solution uses this method in various places to susititue _markdown_ syntax for _HTML_ syntax in the passed-in markdown text.
```

Comment:
Spelling: "Metho" -> "Method" (but see following), "susititue" -> "substitute". There is no **re** _class_, **re.sub** is a function in the **re** module ... confusingly there _is_ a **Pattern.sub** method on the **re.Pattern** class that the module also exposes, but it's usually only used incidentally if you're not creating instances of **re.Pattern** using **re.compile**.

Reply:
> - `re.sub()`:  function of the `re` module that replaces a `regular expression` match with a new value.  The example solutions use this function in various places to substitute _markdown_ syntax for _HTML_ syntax in the passed in markdown text.